app/config/collections/projects.php

```diff
@@ -1848,7 +1859,29 @@
                 'default' => null,
                 'array' => false,
                 'filters' => [],
-            ]
+            ],
+            [
+                '$id' => ID::custom('resourceId'),
```

Comment:
What resourceId and resourceType are we referring to here? How come is this a 1:1 relation if migration can in theory migrate endless resources?

Reply:
these already exists on current migrations. The only new is the `options` attribute.
1. `resourceID` for csv is a `CompoundID` of `databaseId:collectionId`,
2. `resourceType` is Database Group afaik.

Reply:
@abnegate we never use this pattern, let's stick with what we have. I think Matej used a parent resource approach, or we can use a reference approach with having only resource.

Reply:
@eldadfux I can see future issues with the `parentResource` pattern for things like documents/attributes/indexes, which have a 3 level hierarchy: database -> collection -> document, if we use a single attribute but compound as many IDs as we need, it's very flexible. We use it a couple of places in messaging, for referencing bucket/file in a single param. What do you think?

Reply:
I understand, but IIRC we do have a pattern for it already if we want to go with the compound approach, I just want us to be consistent. Please verify how it works with @Meldiron

Reply:
@eldadfux We ended up avoiding "parent" in rules.

What we did, is that rule has:

- `deploymentId` - only set when `type` is `deployment`
- `deploymentResourceType` - can be `function` or `site`
- `deploymentResourceId` - ID of site/function to which deployment belongs

Reply:
After discussion with Darshan, importantto mention this was just addition moved from Cloud. Those already exist on cloud; we just added them to OSS now for this feature